### PR TITLE
Undercut option to duplicate strategy flow

### PR DESCRIPTION
**Pull request by tiagofilipenunes**
- [x] Duplicate modal
- [x] Logic to undercut and copy strategy
- [x] Adjust design to be closer to figma
  - [x] Max modal width is lower than the widget in the figma design
  - [x] Difference in the app button component vs figma component
  - [x] Sheet modal increase height

Questions/additional changes:
- [x] When undercutting, should there be a check that the ranges might overlap and prevent it from happening? No.
- [x] Tooltip when hovering the duplicate option should be updated

### CODE REVIEW

**Comment by ashachaf** [comment]
- [ ] undercup diff is incorrect
the issue is indicating a 0.1% diff, and the implementation is forcing a 1% diff.

File: src/libs/modals/modals/ModalDuplicateStrategy.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+          >
+            <div className="row-span-2 flex h-32 w-32 items-center justify-center self-center rounded-full bg-green/25">
+              <Icon className="h-16 w-16 text-green" />
+            </div>
+            <h3 className="text-14 font-weight-500">{title}</h3>
+            <Button
+              variant="white"
+              type="button"
+              onClick={onClick}
+              aria-label={title.toLowerCase()}
```
Comment (GrandSchtroumpf) [inline src/libs/modals/modals/ModalDuplicateStrategy.tsx:92]: you don't need to set aria-label if the title is already in the h3 above. Some text reader will read it twice, which is annoying for the user.
The only annoying part here is the paragraph below the button, as it'll be read after selecting the button, which is not a good experience. We might want to discuss with Nazar about that.
A solution might be to use `aria-describedBy="descriptionId"` where "descriptionId" is the id of the paragraph below. But I'm not sure about the behavior in a text-reader

**Review by GrandSchtroumpf — APPROVED** [review]
👍

File: src/libs/modals/modals/ModalDuplicateStrategy.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      order1: {
+        ...strategy.order1,
+        startRate: new Decimal(strategy.order1.startRate)
+          .times(1 - undercutDifference)
+          .toString(),
+        endRate: new Decimal(strategy.order1.endRate)
+          .times(1 - undercutDifference)
+          .toString(),
+      },
+    };
```
[the remark this answers is not in the document]
Comment (GrandSchtroumpf) [reply src/libs/modals/modals/ModalDuplicateStrategy.tsx:44]: Maybe you can create a small function to apply the +/- undercutDifference on an amount, but it's not mandatory

[the remark this answers is not in the document]
Comment (tiagofilipenunes) [reply src/libs/modals/modals/ModalDuplicateStrategy.tsx:44]: Agreed - looks more readable with deep clone!

File: src/libs/modals/modals/ModalDuplicateStrategy.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+      description:
+        'Duplicate the strategy with the existing values (price, budget)',
+      testId: 'duplicate-strategy-btn',
+    },
+    {
+      icon: IconCut,
+      title: 'Undercut the Strategy',
+      onClick: undercutStrategy,
+      description:
+        'Set prices at 0.1% tighter spread and try to get filled ahead',
```
Comment (zavelevsky) [inline src/libs/modals/modals/ModalDuplicateStrategy.tsx:69]: don't use "0.1%" - derive it from one place (`undercutDifference`)

**Comment by ashachaf** [comment]
- [ ] undercut option throws an error
image

- [ ] undercutting limit order
could not test, please make sure that when undercutting limit order, there is no issue with 0 price calculation.

**Review by ashachaf — APPROVED** [review]

**Review by zavelevsky — APPROVED** [review]
LGTM

**Pull request merged**